From 01b983c9fcfeea5774ca2df2e167b68c3c299278 Mon Sep 17 00:00:00 2001
From: Jens Axboe <axboe@kernel.dk>
Date: Tue, 19 Nov 2013 18:59:10 -0700
Subject: [PATCH] blk-mq: add blktrace insert event trace
Git-commit: 01b983c9fcfeea5774ca2df2e167b68c3c299278
Patch-mainline: v3.13-rc1
References: fate#315209

We need it to make 'btt' from blktrace happy, otherwise
we are missing one state transition.

Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Jan Kara <jack@suse.cz>

---
 block/blk-mq.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index d5dea24..008f2fb 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -721,6 +721,8 @@ static void __blk_mq_insert_request(struct blk_mq_hw_ctx *hctx,
 {
 	struct blk_mq_ctx *ctx = rq->mq_ctx;
 
+	trace_block_rq_insert(hctx->queue, rq);
+
 	list_add_tail(&rq->queuelist, &ctx->rq_list);
 	blk_mq_hctx_mark_pending(hctx, ctx);
 
-- 
1.6.0.2

